Fix using the Mocha API

KennethSundqvist opened the pull request:
When Mocha is run using the CLI [the default `ui` is first set up](https://github.com/mochajs/mocha/blob/v5.0.5/bin/_mocha#L21), which sets `context.xit`, and [then that setup is overriden](https://github.com/mochajs/mocha/blob/v5.0.5/bin/_mocha#L472) by the `ui` you've specified in the options, so the non-default `ui` can "inherit" parts of the default `ui` by not overriding them.

But when you use the API and construct Mocha with the `ui` option (`const mocha = new Mocha({ui: 'mocha-cakes-2'})`) the default `ui` isn't set up first so Mocha will throw an error [when it runs `exports.xit = context.xit || context.test.skip`](https://github.com/mochajs/mocha/blob/v5.0.5/lib/mocha.js#L207) because mocha-cakes-2 has not set `context.xit` or `context.test`.

iensu commented on test/helpers.js:
```diff
@@ -1,20 +1,88 @@
-  return execFile('mocha', args, { env: testEnv });
+  var func = interfaces[process.env.MOCHA_INTERFACE];
+  if (!func) {
+    throw new Error("The MOCHA_INTERFACE environment variable must be set to one of:", Object.keys(interfaces).join(', '));
```
Making the MOCHA_INTERFACE env variable obligatory will probably break the setup for some people, so I would prefer you to use the `cli` interface if none is specified.

iensu reviewed (changes requested):
Thank you for your PR, this looks really great! One small thing, would be great if you could add a line or two about the MOCHA_INTERFACE env variable in the README. Also see the comment above regarding making it optional.

KennethSundqvist committed:
Add support for the Mocha API.

Mocha throws an error when it runs `exports.xit = context.xit ||
context.test.skip` because mocha-cakes-2 has not set `context.xit` or
`context.test`.

When Mocha is run using the CLI the default ui is set up first, which
sets `context.xit`, and then that setup is overriden by the ui you've
specified in the options, so the non-default ui can "inherit" parts of
the default ui by not overriding them.

But when you use the API and construct Mocha with the ui option (`const
mocha = new Mocha({ui: 'mocha-cakes-2'})`) the default ui isn't set up,
so Mocha runs into that error.

EvilOatmeal commented:
Good points! I've made `cli` the default interface and updated the README.

iensu commented:
Looking good! Thanks a lot!

iensu commented:
PR has been merged and published to NPM under version 3.3.0! Thanks again for your PR!
